src/staking/manager.ts

```diff
@@ -0,0 +1,660 @@
+   */
+  async createWithdrawSlashingBtcTransaction(
+    stakingInput: StakingInputs,
+    stakingParamVersion: number,
+    slashingTx: Transaction,
+    feeRate: number,
+  ): Promise<Transaction> {
+    const p = getBbnParamByVersion(stakingParamVersion, this.stakingParams);
+    if (!p) {
+      throw new Error(`Unable to find staking params for version ${stakingParamVersion}`);
+    }
+
+    const stakerInfo = await this.btcProvider.getStakerInfo();
+    if (!stakerInfo) {
+      throw new Error("Staker info not found while creating withdraw slashing transaction");
+    }
+
+    const staking = new Staking(
+      this.network,
+      stakerInfo,
+      p,
+      stakingInput.finalityProviderPkNoCoordHex,
+      stakingInput.stakingTimelock,
+    );
+
+    const { psbt } = staking.createWithdrawSlashingTransaction(
+      slashingTx,
+      feeRate,
+    );
+
+    const signedSlashingPsbtHex = await this.btcProvider.signPsbt(
+      psbt.toHex()
+    );
+    return Psbt.fromHex(signedSlashingPsbtHex).extractTransaction();
+  }
+
+  private async createBtcDelegationMsg(
+    stakingInstance: Staking,
+    stakingInput: StakingInputs,
+    stakingTx: Transaction,
+    bech32Address: string,
+    stakerInfo: StakerInfo,
+    param: StakingParams,
+    inclusionProof?: btcstaking.InclusionProof,
+  ) {
+    const { transaction: unbondingTx } =
+      stakingInstance.createUnbondingTransaction(stakingTx);
+
+    // Create slashing transactions and extract signatures
+    const { psbt: slashingPsbt } =
+      stakingInstance.createStakingOutputSlashingTransaction(stakingTx);
+    const signedSlashingPsbtHex = await this.btcProvider.signPsbt(
+      slashingPsbt.toHex(),
+    );
+    const signedSlashingTx = Psbt.fromHex(
+      signedSlashingPsbtHex,
+    ).extractTransaction();
+    const slashingSig = extractSchnorrSignaturesFromTransaction(signedSlashingTx);
+    if (!slashingSig) {
+      throw new Error("No signature found in the staking output slashing PSBT");
+    }
+
+    const { psbt: unbondingSlashingPsbt } =
+      stakingInstance.createUnbondingOutputSlashingTransaction(unbondingTx);
+    const signedUnbondingSlashingPsbtHex = await this.btcProvider.signPsbt(
+      unbondingSlashingPsbt.toHex(),
+    );
+    const signedUnbondingSlashingTx = Psbt.fromHex(
+      signedUnbondingSlashingPsbtHex,
+    ).extractTransaction();
+    const unbondingSignatures = extractSchnorrSignaturesFromTransaction(
+      signedUnbondingSlashingTx,
+    );
+    if (!unbondingSignatures) {
+      throw new Error("No signature found in the unbonding output slashing PSBT");
+    }
+
+    // Create Proof of Possession
+    const bech32AddressHex = uint8ArrayToHex(fromBech32(bech32Address).data);
+    const signedBbnAddress = await this.btcProvider.signMessage(
+      bech32AddressHex,
+      "ecdsa",
+    );
+    const ecdsaSig = Uint8Array.from(Buffer.from(signedBbnAddress, "base64"));
+    const proofOfPossession: ProofOfPossessionBTC = {
```

Comment:
Imo it would be nice if proof of possession was up to the caller i.e it should be up to the user of the library which pop to use, but I understand this is due to the fact that we weren't able to make bip322 work against Babylon ? (at least from FE)